wallet/src/main/java/com/android/identity_credential/wallet/NfcEngagementHandler.kt

```diff
@@ -152,6 +162,10 @@ class NfcEngagementHandler : HostApduService() {
             if (engagementHelper != null) {
                 Logger.w(TAG, "Reader didn't connect inside $timeoutSeconds seconds, closing")
                 engagementHelper!!.close()
+
+                if (PresentationActivity.isPresentationActive()) {
```

Comment:
I think this logic check can be encapsulated within the function `PresentationActivity.stopPresentationReaderTimeout()` that should always get called when `onDeactivated` gets called. 

i.e. always call `PresentationActivity.stopPresentationReaderTimeout()` and perform the actual operation of "stopping the reader timeout" if needed, inside that class.

Reply:
For example, just like `engagementDetected()`, the internal state `Phase` is not checked outside of that function
```
 fun engagementDetected(context: Context) {
            if (phase.value != Phase.NOT_CONNECTED) {
                Logger.w(TAG, "nfcEngagementDetected: expected NOT_CONNECTED, is in " + phase.value)
                return
            }
```
but in this case during `onDeactivated()`, the internal `PresentationActivity`'s` `Phase` is exposed through a public function
```
        fun isPresentationActive(): Boolean {
            return phase.value != Phase.NOT_CONNECTED
        }
```
 
and this check is performed outside of `PresentationActivity` only so the logic flow can lead back to `PresentationActivity`